### ISSUE TITLE
 insufficient log options when there is a bad /etc/crypttab entry

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#12272]
This is copied verbatim from the downstream bug report at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=920234

Package: systemd
Version: 240-4
Severity: normal

When I have a bogus /etc/crypttab entry (for example when converting an
encrypted laptop to a virtual machine that's not encrypted) the boot process
hangs for 90 seconds for no obvious reason.
```
[    **] A start job is running for /dev/dis·9597-dbdb2fedbd1f (24s / 1min 30s)
```
When seeing the above anyone who doesn't know that "grep -R dbdb2fedbd1f /etc"
is necessary will have trouble determining the cause of this.
```
# systemd-analyze blame
Bootup is not yet finished (org.freedesktop.systemd1.Manager.FinishTimestampMonotonic=0).
Please try again later.
Hint: Use 'systemctl list-jobs' to see active jobs
# systemctl list-jobs
JOB UNIT                                                           TYPE  STATE  
181 dev-disk-by\x2duuid-…75\x2d410c\x2d9597\x2ddbdb2fedbd1f.device start running
179 systemd-cryptsetup@nvme0n1p2_crypt.service                     start waiting

2 jobs listed
```
The above commands run during the boot process makes the problem obvious.  But
what if the cryptsetup command completely fails before the sysadmin logs in?
```
# systemd-analyze blame|head
          1.051s ifupdown-wait-online.service
           958ms tor@default.service
           753ms postfix@-.service
           494ms udisks2.service
           480ms networking.service
           445ms dev-vda.device
           260ms accounts-daemon.service
           245ms gdomap.service
           217ms systemd-timesyncd.service
           216ms auditd.service
```
I would hope to see a 90 second entry in the above, but we don't get one.
```
# systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @1min 31.348s
└─multi-user.target @1min 31.347s
  └─postfix.service @1min 31.339s +7ms
    └─postfix@-.service @1min 30.583s +753ms
      └─basic.target @1min 30.512s
        └─sockets.target @1min 30.506s
          └─dbus.socket @1min 30.505s
            └─sysinit.target @1min 30.492s
              └─systemd-update-utmp.service @902ms +41ms
                └─auditd.service @664ms +216ms
                  └─systemd-tmpfiles-setup.service @602ms +50ms
                    └─local-fs.target @598ms
                      └─run-user-0.mount @1min 45.931s
                        └─swap.target @940ms
                          └─dev-vdb.swap @909ms +27ms
                            └─dev-vdb.device @876ms
```
I expect to see something with +90s in the above, but it doesn't happen.
Something between systemd-update-utmp.service and sysinit.target took 90
seconds and there's no obvious way of discovering what it was.

**Comment by poettering** [comment]
The "systemd-analyze" output only shows data about activated units. Units that are never activated don't show up there, and that's the problem here: the device unit never showed up.

If you want timing info for what systemd does and waits for, then please use system logs (possibly with debug logging enabled), but "systemd-analyze blame" and "systemd-analyze critical-chain" are really not the tools for that. Those tools assume you know exactly what you do, they are debugging tools for hackers if you so will, and those are folks who understand they only show info about started data. In fact the legend in the critical chain output even makes that clear by saying "active or started", but these .device units obviously never did.

Hence, this is a documentation issue I guess to some point, but not much else.

(Showing info about units that never activated here is problematic since it would mean we would have to keep data in memory about anything we did so far which could grow without bounds. That's not how systemd operates though: it agressively GC's stuff from memory and only keeps stuff in memory that are started, shall change state or are referenced directly or indirectly by that, and I am very sure we shouldn't change that and let memory use grow uncontrolled.

Hence: use logs where logs are right

**Comment by poettering** [comment]
Fix waiting in #13036
